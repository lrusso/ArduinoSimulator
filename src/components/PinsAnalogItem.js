import React from "react"

const PinsAnalogItem = ({ value }) => {
  return <div style={styles.container}>{value}</div>
}

const styles = {
  container: {
    float: "left",
    width: "40px",
    height: "16px",
    borderRadius: "16px",
    marginTop: "4px",
    marginLeft: "4px",
    backgroundColor: "black",
    fontSize: "11px",
    fontFamily: "Arial",
    color: "white",
    justifyContent: "center",
    alignItems: "center",
    display: "flex",
    cursor: "default",
  },
}

export default PinsAnalogItem
