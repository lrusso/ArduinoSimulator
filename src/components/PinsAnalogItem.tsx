import React from "react";
import { Gpio_Analog } from "../utils/interfaces";

import { setAnalogPin } from "../utils/interpreter";
import { useSimulatorContext } from "../contexts/SimulatorContext";

interface PinsAnalogItemProps {
  gpioAnalog: Gpio_Analog;
}

const PinsAnalogItem = ({ gpioAnalog }: PinsAnalogItemProps) => {
  const { handleSetDigitalPins } = useSimulatorContext();

  const handler_onchange = (e) => {
    const newValue = e.target.value;
    gpioAnalog.duty = e.target.value;
    setAnalogPin(gpioAnalog.pinNumber, newValue);
    handleSetDigitalPins(gpioAnalog.pinNumber, newValue);
  };

  return (
    <input
      type="number"
      style={styles.input}
      value={gpioAnalog.duty}
      onChange={handler_onchange}
      disabled={!gpioAnalog.isInput}
    ></input>
  );
};

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    float: "left",
    width: "40px",
    height: "16px",
    borderRadius: "16px",
    marginTop: "4px",
    marginLeft: "4px",
    backgroundColor: "black",
    fontSize: "11px",
    fontFamily: "Arial",
    color: "white",
    justifyContent: "center",
    alignItems: "center",
    display: "flex",
    cursor: "default",
  },
  input: {
    display: "inline-block",
    background: "#333",
    margin: "2px",
    border: "none",
    width: "40px",
    color: "white",
  },
};

export default PinsAnalogItem;
