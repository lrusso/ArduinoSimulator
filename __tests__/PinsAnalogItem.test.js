import React from "react"
import renderer from "react-test-renderer"
import PinsAnalogItem from "../src/components/PinsAnalogItem"

describe("PinsAnalogItem", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<PinsAnalogItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
