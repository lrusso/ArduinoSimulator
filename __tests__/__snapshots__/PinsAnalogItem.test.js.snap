// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinsAnalogItem Rendering 1 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": "16px",
      "color": "white",
      "cursor": "default",
      "display": "flex",
      "float": "left",
      "fontFamily": "Arial",
      "fontSize": "11px",
      "height": "16px",
      "justifyContent": "center",
      "marginLeft": "4px",
      "marginTop": "4px",
      "width": "40px",
    }
  }
/>
`;
